test/enums.jl

```diff
@@ -0,0 +1,130 @@
+module TestEnums
+
+using Base.Test
+
+@test_throws MethodError convert(Base.Enums.Enum, 1.0)
+
+Base.Enums.@enum Fruit apple orange kiwi
```

Comment:
Wouldn't it make more sense to import the `@enums` macro and then use it bare? I'd expect that we export this from Base if this gets merged.